### ISSUE TITLE
PR build with depends doesn't show the correct sha

### ISSUE THREAD

**Issue by pshipton** [eclipse/openj9#9537]
@AdamBrousseau PR testing with the `depends eclipse/omr#master` show incorrect shas. Looking at the top level build https://ci.eclipse.org/openj9/job/PullRequest-OpenJ9/3544/ it shows `OMR: cd259de`, which was the openj9 branch head, and not what was actually tested. If you dig into one of the builds https://ci.eclipse.org/openj9/job/Build_JDK8_s390x_linux_Personal/626, you can see it actually built from `HEAD is now at b8634d51d`, which was the head of the master branch at the time.

**Comment by pshipton** [comment]
I'm also wondering if all the builds are guaranteed to use the same OMR sha in this case, or would they get master at the time each build ran.

**Comment by AdamBrousseau** [comment]
Yes this has always been a caveat of the PR depends. And yes, if you're kicking several builds and they end up staggering you could end up testing a different Sha in each. This is because the depends logic is down in the compile job instead of at the top level. It could be refactored but I'm not sure how much with that would be. I'd have to take a look.
